packages/teams-js/src/internal/videoPerformanceMonitor.ts

```diff
@@ -0,0 +1,119 @@
+import { VideoFrameTick } from './videoFrameTick';
+import { VideoPerformanceStatistics } from './videoPerformanceStatistics';
+
+export class VideoPerformanceMonitor {
+  private static readonly distributionBinSize = 1000;
+  private static readonly calculateFPSInterval = 1000;
+
+  private isFirstFrameProcessed = false;
+
+  // The effect that the user last selected:
+  private applyingEffect: {
+    effectId: string;
+    effectParam?: string;
+  };
+
+  // The effect that is currently applied to the video:
+  private appliedEffect: {
+    effectId: string;
+    effectParam?: string;
+  };
+
+  private frameProcessTimeLimit = 100;
+  private gettingTextureStreamStartedAt: number;
+  private currentStreamId: string;
+  private frameProcessingStartedAt = 0;
+  private frameProcessingTimeCost = 0;
+  private processedFrameCount = 0;
+
+  private performanceStatistics: VideoPerformanceStatistics;
+
+  public constructor(private reportPerformanceEvent: (actionName: string, args: unknown[]) => void) {
+    this.performanceStatistics = new VideoPerformanceStatistics(VideoPerformanceMonitor.distributionBinSize, (result) =>
+      this.reportPerformanceEvent('video.performance.performanceDataGenerated', [result]),
+    );
+  }
+
+  public startMonitorSlowFrameProcessing(): void {
```

Comment:
Not required for me to approve since this is an internal class, but comments about these public functions will help people in the future.

Reply:
Added comments to the whole class